### PR TITLE
Beautify Code + plus minor tweaks

### PR DESCRIPTION
**Pull request by vramana**
I am happy to explain any changes and revert any changes that you feel are unnecessary.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -13,8 +13,5 @@ module.exports = {
       loader: 'babel-loader'
     }]
   },
-  resolve: {
-    extensions: ['', '.js']
-  },
```
Comment (nrc) [inline webpack.config.js]: Why do we not need this?

File: static/snippet.js [hunk 2]
```diff
@@ -3,15 +3,16 @@ import ReactDOM from 'react-dom';
 
 class Snippet extends React.Component {
     render() {
-        if (!this.props.spans || this.props.spans.length == 0) {
+        const { sans } = this.props;
```
Comment (nrc) [inline static/snippet.js]: typo here - I expect this should be `spans` instead of `sans` and `this.props.spans`, not `this.props`?

Comment (nrc) [inline static/snippet.js]: er, no, just the first one, I think

**Comment by nrc** [comment]
Thank you! That's very useful!

File: webpack.config.js (hunk 1 above)
Comment (vramana) [inline webpack.config.js]: It was required in Webpack 1 not in Webpack 2. https://webpack.js.org/guides/migrating/#resolve-extensions

Comment (nrc) [inline webpack.config.js]: cool, thanks!

**Comment by nrc** [comment]
@vramana do you have the input file for yarn please? I'm not sure how to generate the yarn.lock file.

**Comment by vramana** [comment]
You jto need to run `yarn` in the command line. It will generate `yarn.lock` file. If there are any issues, I will send a PR 😄

**Comment by nrc** [comment]
huh, clever, I will give it a go, thanks!

**Comment by vramana** [comment]
Wait. Since I already committed the `yarn.lock` file it shouldn't complain about it again. Can you paste the error you are getting the next time you see it?

**Comment by nrc** [comment]
I haven't tried using Yarn yet. I added another dependency and I assumed I would have to add it to a Yarn config file. But it sounds like Yarn can figure that out without a config file.

**Pull request merged**